crates/codegen/syntax/src/combinator_node.rs

```diff
@@ -42,6 +42,7 @@ pub enum CombinatorNode<'context> {
         primary_expression: &'context CombinatorNode<'context>,
     },
     Reference {
+        name: Option<String>,
```

Comment:
I'm not sure of the use case for renaming references. The referenced tree is already named. I thought this issue was about eliminating the _* names, which would be done by adding names to e.g. repetitions?

Reply:
> I'm not sure of the use case for renaming references.

The current PR is about #339, which makes sure we can give names to choices + references.

> I thought this issue was about eliminating the _* names, which would be done by adding names to e.g. repetitions?

That would be #353, which is still WIP.

For references, it is needed to give names to inlined nodes (that will not generate their own `NodeKind`). Please check the description of #323 for some examples.

Reply:
Will remove the changes to `reference` and keep `choice`.